e2e/Tests/iothub/service/FileUploadNotificationE2ETest.cs

```diff
@@ -119,48 +119,34 @@ async Task<AcknowledgementType> OnFileUploadNotificationReceived(FileUploadNotif
+            // arrange
+            using var sender = new IotHubServiceClient(TestConfiguration.IotHub.ConnectionString);
+            bool connectionLossEventExecuted = false;
+            Func<ErrorContext, Task> OnConnectionLost = delegate
             {
-                Protocol = protocol
+                // There is a small chance that this test's connection is interrupted by an actual
+                // network failure (when this callback should be executed), but the operations
+                // tested here are so quick that it should be safe to ignore that possibility.
+                connectionLossEventExecuted = true;
+                return Task.CompletedTask;
             };
+            sender.FileUploadNotifications.ErrorProcessor = OnConnectionLost;
 
-            using var serviceClient = new IotHubServiceClient(TestConfiguration.IotHub.ConnectionString, options);
-
-            try
-            {
-                var errorProcessorNotified = new TaskCompletionSource<bool>(TaskCreationOptions.RunContinuationsAsynchronously);
-                serviceClient.FileUploadNotifications.FileUploadNotificationProcessor = (_) => Task.FromResult(_defaultAcknowledgementType);
-                serviceClient.FileUploadNotifications.ErrorProcessor = (errorContext) =>
-                {
-                    VerboseTestLogger.WriteLine("Error processor fired.");
-                    errorProcessorNotified.TrySetResult(true);
-                    return Task.CompletedTask;
-                };
-
-                VerboseTestLogger.WriteLine("Opening client...");
-                await serviceClient.FileUploadNotifications.OpenAsync().ConfigureAwait(false);
-                VerboseTestLogger.WriteLine("Client opened.");
-
-                VerboseTestLogger.WriteLine("Client closing...");
-                await serviceClient.FileUploadNotifications.CloseAsync().ConfigureAwait(false);
-                VerboseTestLogger.WriteLine("Client closed.");
-
-                // The open file upload notification processor should be able to receive more than one
-                // file upload notification without closing and re-opening as long as there is more
-                // than one file upload notification to consume.
-                using var cts = new CancellationTokenSource(TimeSpan.FromMilliseconds(TestTimeoutMilliseconds));
-                await errorProcessorNotified.WaitAsync(cts.Token).ConfigureAwait(false);
-                errorProcessorNotified.Task.IsCompleted.Should().BeTrue();
-            }
-            finally
-            {
-                serviceClient.FileUploadNotifications.ErrorProcessor = null;
-                await serviceClient.FileUploadNotifications.CloseAsync().ConfigureAwait(false);
+            Task<AcknowledgementType> OnFileUploadNotificationReceivedAsync(FileUploadNotification fileUploadNotification) 
+            { 
+                return Task.FromResult(AcknowledgementType.Complete); 
```

Comment:
If you accept them here, other tests that are expecting one may not get it. Better to reject?

Reply:
Good point.